### PR TITLE
2FA robustness improvements

### PR DESCRIPTION
**Pull request by vgrichina**
This is expected to make 2FA requests a bit faster (do not block for delete) and more robust (@mattlockyer's change fixes race condition).

### CODE REVIEW

**Comment by vgrichina** [comment]
@mehtaphysical this works fine with https://github.com/near/near-wallet/pull/1706

File: src/account_multisig.ts [hunk 1]
```diff
@@ -53,36 +44,63 @@ export class AccountMultisig extends Account {
...
+                return await this.signAndSendTransaction(receiverId, actions);
+            }
+            throw e;
+        }
+
+        // TODO: Are following even needed? Seems like it throws on error already
+        if (!result.status) {
+            throw new Error('Request failed');
+        }
+        const status: any = Object.assign({}, result.status);
```
Comment (mehtaphysical) [inline src/account_multisig.ts:71]: nit:

```suggestion
        const status: any = { ...result.status };
```

File: src/account_multisig.ts [hunk 2]
```diff
@@ -209,9 +220,6 @@ export class Account2FA extends AccountMultisig {
     async sendCodeDefault() {
         const { accountId } = this;
         const { requestId, actions } = this.getRequest();
```
Comment (mehtaphysical) [inline src/account_multisig.ts:222]: Fix linting:

```suggestion
        const { requestId } = this.getRequest();
```

**Review by mehtaphysical — APPROVED** [review]
Looks good. One nit and one linting change to fix tests.

Also looks like there is a merge conflict

**Comment by mehtaphysical** [comment]
@vgrichina I forgot, there was a change so we require 2 approvals now

**Comment by vgrichina** [comment]
> @vgrichina I forgot, there was a change so we require 2 approvals now

but why?

**Comment by mehtaphysical** [comment]
@mikedotexe can probably add more context about this change.

**Pull request merged**